src/client/src/tscSQLParser.c

```diff
@@ -656,11 +648,12 @@ int32_t tscValidateSqlInfo(SSqlObj* pSql, struct SSqlInfo* pInfo) {
       break;
     }
 
+    case TSDB_SQL_DROP_DB:
     case TSDB_SQL_USE_DB: {
       const char* msg = "invalid db name";
       SStrToken* pToken = taosArrayGet(pInfo->pMiscInfo->a, 0);
 
-      if (tscValidateName(pToken, false, NULL) != TSDB_CODE_SUCCESS) {
+      if (tscValidateName(pToken, true, NULL) != TSDB_CODE_SUCCESS) {
```

Comment:
Escaping with the backward "`" quotations is turned on with "escapeEnabled=true".